chore(release): v2.5.2-beta

shamardy opened the pull request:
## v2.5.2-beta - 2025-10-10

### Enhancements/Fixes:

**Swap Stats DB**:
- Swap status broadcasting was enabled for privacy coins with the persistent pubkey hidden (set to zeros) to maintain user privacy. [#2648](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2648)
- Stats database now correctly stores persistent pubkeys for both maker and taker instead of incorrectly storing htlc pubkeys in some cases. [#2648](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2648)

**ARRR/Pirate**:
- The `get_nullifiers` function for Zcoin WASM build is now aligned with its sqlite counterpart to return nullifiers for both unspent notes and notes with unconfirmed spends, fixing the `spent_by_me` field in transaction history and balance calculations. [#2651](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2651)

**Metrics**:
- The `memory_db` size metric that relied on `parity-util-mem::malloc_size` was removed because it intermittently segfaulted on Linux due to allocator conflicts. [#2632](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2632)

**Build and Dependency Management**:
- A CI job was added to build macOS Universal2 artifacts for KDF, this combines `x86_64-apple-darwin` and `aarch64-apple-darwin` binaries via `lipo` to produce a single binary that runs natively on both Intel and Apple Silicon. The universal binary is uploaded as `kdf_<commit>-mac-universal.zip`. [#2628](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2628)
- The `parity-util-mem` dependency was removed. [#2632](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2632)
- CI container base image was bumped to `debian:bullseye-slim` [#2534](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2534) [#2641](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2641)

shamardy committed:
Merge pull request #2554 from KomodoPlatform/main

chore(release): propagate `get_private_keys` hotfix (main→staging)

shamardy committed:
fix(metrics): remove memory_db size metric (#2632)

`malloc_size` caused a segfault in linux and was removed.

Commit:
feat(ci): add macos universal2 build (#2628)

shamardy committed:
chore(release): bump mm2 version to 2.5.2-beta (#2638)

shamardy committed:
chore(release): add changelog entries for v2.5.2-beta (#2639)

v2.5.1-beta changelogs were added as well.

shamardy committed:
chore(ci): bump CI container image to debian bullseye-slim to match dev (#2641)

This change is re-propagated to dev only to preserve linear git history

smk762 commented:
Confirmed related gui login bug on linux is resolved with this build.
Awaiting GUI Mac binaries to test confirmation of universal binaries launching successfully on pre-silicon devices

smk762 commented:
Confirmed KDF binary in `kdf_8beb119-mac-universal.zip` via CI sdk builds launches successfully on a pre-silicon MacMini in terminal. Still having issues with GUI, but does not appear to be related to KDF.

Commit:
fix(zcoin): align `get_nullifiers` with sqlite counterpart (#2651)

This commit is a quick fix for zcoin wasm build to return correct balance and tx history (spent_amount).
It corrects the `get_nullifiers` function to return nullfiers both for unspent notes and notes with unconfirmed spends similar to the non-wasm sqlite function in librustzcash.

mariocynicys committed:
feat(legacy-stats): broadcast swap status for privacy coins (#2648)

This commit hides the persistent pubkey while broadcasting swap status for privacy coins. It also fixes a problem where the htlc pubkey where stored in stats db in some cases instead of the persistent pubkey.

shamardy committed:
chore(release): finalize changelog for v2.5.2-beta (#2659)

smk762 reviewed (approved):
LGTM thanks!
